File: backend/apps/main/urls.py
from django.urls import path, include

from main.views.proxy import ProxyView
from main.views.ws import AuthParamsView
from main.views import ftx
from main.views.orders import OrdersListView, CancelAllOrdersView, CancelOrderView
from main.views.trade import TradeView, TradeDetailView, CancelTradesView, MarketOrderView, LimitOrderView


urlpatterns = [
    path('trades/cancel/<str:exchange>/', CancelTradesView.as_view(), name='trades-cancel'),
    path('trades/<int:pk>/', TradeDetailView.as_view(), name='trade-detail'),
    path('trades/<str:exchange>/', TradeView.as_view(), name='trade'),
    path('ws-auth-params/<str:exchange>/', AuthParamsView.as_view(), name='auth-params'),

    path('orders/cancel/<int:order_id>', CancelOrderView.as_view(), name='cancel-all-orders'),
    path('orders/<str:symbol>/', OrdersListView.as_view(), name='open-orders-list'),

    path('market/', MarketOrderView.as_view(), name='market'),
    path('limit/', LimitOrderView.as_view(), name='limit'),

    path('huobi/orders/cancel/', CancelAllOrdersView.as_view(), name='huobi-cancel-all-orders'),

    path('ftx/', include([
        path('symbols/', ftx.SymbolsListView.as_view(), name='symbols'),
        path('balances/', ftx.BalancesListView.as_view(), name='balances'),
        path('account/', ftx.AccountDetailView.as_view(), name='account-detail'),
        path('positions/', ftx.PositionsListView.as_view(), name='positions'),
        path('positions/market/', ftx.PositionMarketOrderView.as_view(), name='close-position'),
        path('orders/open/', ftx.OpenOrdersListView.as_view(), name='open-orders'),
        path('fills/', ftx.FTXFillsView.as_view(), name='fills'),
        path('orders/cancel/<int:id>/', ftx.CancelOrderView.as_view(), name='cancel-orders'),
        path('orders/modify/<int:id>/', ftx.ModifyOrderView.as_view(), name='modify-orders'),
        path('trigger-orders/<str:market>/', ftx.TriggerOrdersView.as_view(), name='open-trigger-orders'),
        path('twap-orders/<str:market>/', ftx.TWAPOrdersView.as_view(), name='active-twap-orders'),
        path('orders/cancel/', ftx.CancelAllOrdersView.as_view(), name='ftx-cancel-all-orders'),
        # path('place/', ftx.PlaceFTXOrderView.as_view(), name='place'),
    ])),
    path('proxy/', ProxyView.as_view(), name='proxy'),
]

File: frontend/src/urls.js
export const domain = window.origin.replace('3000', '8000') // https://hftcryptobot.com
export const wsDomain = domain.replace('http', 'ws')

export const SIGNIN = '/users/sign-in'
export const USER_DETAIL = '/users/user'
export const USER_SETTINGS = '/users/user_settings'
export const SIGNUP = '/users/sign-up'
export const CONFIRM = '/users/confirm'
export const RESET_PASSWORD = '/users/reset_password'

export const TRADE = '/main/trades/{exchange}/'
export const BALANCE = '/main/ws-auth-params/{exchange}/'
export const OPEN_ORDERS = '/main/orders/{symbol}/'
export const CANCEL_ALL_ORDERS = '/main/huobi/orders/cancel/'
export const CANCEL_ORDER = '/main/orders/cancel/{id}'
export const TRADE_DETAIL = '/main/trades/{id}/'
export const CANCEL_TRADES = '/main/trades/cancel/{exchange}/'
export const MARKET = '/main/market/'
export const LIMIT = '/main/limit/'

export const HUOBI_DOMAIN = 'https://api.huobi.pro'
export const HUOBI_SYMBOLS = `${HUOBI_DOMAIN}/v2/settings/common/symbols/`
export const HUOBI_KLINES = `${HUOBI_DOMAIN}/market/history/kline?symbol={symbol}&size={size}&period={period}`
export const HUOBI_SYMBOL_SETTINGS = `${HUOBI_DOMAIN}/v1/settings/common/symbols`

export const FTX_SYMBOLS = 'main/ftx/symbols/'
export const FTX_POSITIONS_LIST = 'main/ftx/positions/'
export const CLOSE_POSITION_MARKET = 'main/ftx/positions/market/'
export const FTX_OPEN_ORDERS_LIST = 'main/ftx/orders/open/'
export const FTX_OPEN_TRIGGER_ORDERS_LIST = 'main/ftx/trigger-orders/{symbol}/'
export const FTX_ACTIVE_TWAP_ORDERS_LIST = 'main/ftx/twap-orders/{symbol}/'
export const FTX_BALANCES = 'main/ftx/balances/'
export const FTX_ACCOUNT = 'main/ftx/account/'
export const FTX_CANCEL_ORDER = 'main/ftx/orders/cancel/{id}/'
export const FTX_MODIFY_ORDER = 'main/ftx/orders/modify/{id}/'
export const PROXY_API = 'main/proxy/'
export const FTX_FILLS_LIST = 'main/ftx/fills/'
export const FTX_CANCEL_ALL_ORDERS = '/main/ftx/orders/cancel/'
// export const FTX_PLACE_ORDER = 'main/ftx/place/'

export const LOGS_WS = `${wsDomain}/api/v1/logs/{id}/`
